query_execution/QueryExecutionState.hpp

```diff
@@ -92,8 +92,15 @@ class QueryExecutionState {
                                const std::size_t num_rebuild_workorders,
                                const bool rebuild_initiated) {
     DCHECK(operator_index < num_operators_);
-    rebuild_status_[operator_index].second = num_rebuild_workorders;
-    rebuild_status_[operator_index].first = rebuild_initiated;
+    auto search_res = rebuild_status_.find(operator_index);
+    if (search_res != rebuild_status_.end()) {
+      search_res->second.has_initiated = rebuild_initiated;
+      search_res->second.num_pending_workorders = num_rebuild_workorders;
+    } else {
+      RebuildStatus rebuild_status(rebuild_initiated, num_rebuild_workorders);
+
+      rebuild_status_.emplace(operator_index, std::move(rebuild_status));
```

Comment:
I am trying to understand the difference between the ``std::move`` usage here and if you use ``std::foward_as_tuple`` instead. For instance, line 100 can be removed and instead line 102 will look as follows:

``rebuild_status_.emplace(operator_index, std::std::forward_as_tuple(rebuild_initiated, num_rebuild_workorders));``

Do you think any one version is better than another?